src/core/src/package_managers/YumPackageManager.py

```diff
@@ -325,26 +325,126 @@ def extract_dependencies(self, output, packages):
+        # Package polkit-0.115-13.el8_5.2.x86_64 is already installed.
+        # Dependencies resolved.
+        # ================================================================================
+        # Package   Arch   Version          Repository                              Size
+        # ================================================================================
+        # Upgrading:
+        # polkit    x86_64 0.115-14.el8_6.1 rhel-8-for-x86_64-baseos-eus-rhui-rpms 154 k
+        # polkit-libs
+        #           x86_64 0.115-14.el8_6.1 rhel-8-for-x86_64-baseos-eus-rhui-rpms  77 k
+        #
+        # Transaction Summary
+        # ================================================================================
+        # Upgrade  2 Packages
+        #
+        # Total download size: 231 k
+        # Operation aborted.
+
         dependencies = []
+        package_arch_to_look_for = ["x86_64", "noarch", "i686"]
+
         lines = output.strip().split('\n')
 
-        for line in lines:
-            if line.find(" will be updated") < 0 and line.find(" will be an update") < 0 and line.find(" will be installed") < 0:
-                self.composite_logger.log_debug(" - Inapplicable line: " + str(line))
-                continue
+        for line_index in range(0, len(lines)):
+            line = re.split(r'\s+', (lines[line_index].replace("--->", "")).strip())
+            next_line = []
+            dependent_package_name = ""
+
+            if line_index < len(lines) - 1:
+                next_line = re.split(r'\s+', (lines[line_index + 1].replace("--->", "")).strip())
 
-            updates_line = re.split(r'\s+', line.strip())
-            if len(updates_line) != 7:
+            if self.is_valid_update(line, package_arch_to_look_for):
+                dependent_package_name = self.get_product_name_with_arch(line, package_arch_to_look_for)
+            elif self.is_valid_update(line+next_line, package_arch_to_look_for):
+                dependent_package_name = self.get_product_name_with_arch(line+next_line, package_arch_to_look_for)
+            else:
                 self.composite_logger.log_debug(" - Inapplicable line: " + str(line))
                 continue
 
-            dependent_package_name = self.get_product_name(updates_line[2])
-            if len(dependent_package_name) != 0 and dependent_package_name not in packages:
+            if len(dependent_package_name) != 0 and dependent_package_name not in packages and dependent_package_name not in dependencies:
                 self.composite_logger.log_debug(" - Dependency detected: " + dependent_package_name)
                 dependencies.append(dependent_package_name)
 
         return dependencies
 
+    def is_valid_update(self, package_details_in_output, package_arch_to_look_for):
+        return len(package_details_in_output) == 6 and self.is_arch_in_package_details(package_details_in_output[1], package_arch_to_look_for)
```

Comment:
why it's a 6 here, can we use a constant for this?

Reply:
Demonstrate why it's six with a comment showing a sample as function comment

Reply:
Done.
Added a comment to explain why length should be a 6 here. Not adding as a constant since it's scope is limited to this function only